From 329f497fec66875e5a2ed7e82b91f12a0289484e Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:06:58 -0400
Subject: [powerpc] eeh: More accurate log

Message-id: <1410545655-205645-190-git-send-email-dzickus@redhat.com>
Patchwork-id: 94277
O-Subject: [RHEL7 PATCH 189/626] powerpc/eeh: More accurate log
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 0b5381a61821d1c074c76e01b27bf5f69cf2dd01
Author: Gavin Shan <shangw@linux.vnet.ibm.com>
Date:   Tue Nov 12 14:49:22 2013 +0800

    powerpc/eeh: More accurate log

    This clarifies in the log whether the error is a global PHB error
    or an individual PE being frozen.

    Signed-off-by: Gavin Shan <shangw@linux.vnet.ibm.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/eeh_event.c b/arch/powerpc/kernel/eeh_event.c
index d27c5af..72d748b 100644
--- a/arch/powerpc/kernel/eeh_event.c
+++ b/arch/powerpc/kernel/eeh_event.c
@@ -74,8 +74,13 @@ static int eeh_event_handler(void * dummy)
   pe = event->pe;
   if (pe) {
    eeh_pe_state_mark(pe, EEH_PE_RECOVERING);
-   pr_info("EEH: Detected PCI bus error on PHB#%d-PE#%x\n",
-     pe->phb->global_number, pe->addr);
+   if (pe->type & EEH_PE_PHB)
+    pr_info("EEH: Detected error on PHB#%d\n",
+      pe->phb->global_number);
+   else
+    pr_info("EEH: Detected PCI bus error on "
+     "PHB#%d-PE#%x\n",
+     pe->phb->global_number, pe->addr);
    eeh_handle_event(pe);
    eeh_pe_state_clear(pe, EEH_PE_RECOVERING);
   } else {
-- 
1.7.1